lemur/sources/service.py

```diff
@@ -152,6 +167,8 @@ def sync_certificates(source, user):
             cert = parse_certificate(certificate["body"])
             matching_serials = certificate_service.get_by_serial(serial(cert))
             exists = find_matching_certificates_by_hash(cert, matching_serials)
+            # create a mapping for mis-matched certificates due to different names
+            certs_with_mismatched_name[certificate["name"]] = [x for x in exists if x]
```

Comment:
I'm not familiar with this python syntax; do we need special handling for the case where it's still not found even by hash?

Reply:
`find_matching_certificates_by_hash ` returns a list, and in this case`[x for x in exists if x]` would work. However, I agree, it is more safe to  make sure exists is a list